code-examples/smart_alert_setting.py

```diff
@@ -0,0 +1,202 @@
+              format(api_response.reason, URL),
+              file=stderr)
+        return []
+
+    # convert the response to JSON and
+    # get the 'data' element
+    api_response = api_response.json()
+    data = api_response['data']
+
+    return data.get('id', None)
+
+
+def get_triggered_smart_alerts(leader, key, smart_alert_setting_id,
+                               triggered_smart_alerts=[], page=1):
+    """Get the smart alert triggered with the given smart alert setting id
+
+    This is in this example to show that the smart alert setting was created
+        with the correct attributes.
+
+    Args:
+        leader(string): box to use to create the smart alert setting
+        key(string): an API token granting permission to POST a smart alert on
+            the given leader.
+        smart_alert_setting_id(string): id of the smart alert setting to get
+        triggered_smart_alerts(list(string)): ids of smart alerts triggered by
+            the given smart alert setting
+        page(int): current page of alerts
+
+    Returns:
+        list(string): the ids of the triggered alerts
+    """
+    # Python quirk
+    if triggered_smart_alerts is None:
+        triggered_smart_alerts = []
+
+    # build the URL, filtering on only smart alerts (of type smart_thresh)
+    URL = ('https://{leader}/api/sp/alerts/?'
+           'filter=/data/attributes/alert_type=smart_thresh&'
+           'page={page}').format(
+        leader=leader,
+        resource=smart_alert_setting_id,
+        page=page)
+
+    # make the API request
+    api_response = requests.get(
+        URL,
+        headers={
+            'X-Arbux-APIToken': key
+        },
+        verify=CERT_FILE)
+
+    # check the API response code
+    if api_response.status_code != requests.codes.ok:
+        print("[GET ERROR] API responded with this error: "
+              "{}\n(url: {})".
+              format(api_response.reason, URL),
+              file=stderr)
+        return None
+
+    # convert the response to JSON and get the 'data' element
+    api_response = api_response.json()
+    data = api_response['data']
+    if data == []:
+        # No more results
+        return triggered_smart_alerts
+
+    for resource in data:
+        if ('relationships' in resource and
+                'smart_alert_setting' in resource['relationships']):
+            id_ = (resource['relationships']
+                   ['smart_alert_setting']['data']['id'])
+            if id_ == smart_alert_setting_id:
+                triggered_smart_alerts.append(resource['id'])
+
+    # iterate over pages of alerts to find the triggered alert with the given
+    # smart alert setting
+    page += 1
+    return get_triggered_smart_alerts(
+        leader, key, smart_alert_setting_id, triggered_smart_alerts, page)
+
+
+if __name__ == '__main__':
+    #
+    # set the SP leader hostname and API key
+    #
+    SP_LEADER = 'leader.example.com'
+    API_KEY = 'eFvokphdyGHA_M4oLlLtfDnlIf9bpjFnn0mWlDqw'
```

Comment:
Some of our examples leave the key in, some use `API_KEY = os.getenv('API_TOKEN')`, some get it from cmd line args.  This is fine with me, but in case you didn't like having the key in there, these are some options.